src/libraries/System.Net.Http.Json/ref/System.Net.Http.Json.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the https://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System.Net.Http.Json
+{
+    public static partial class HttpClientJsonExtensions
+    {
+        public static System.Threading.Tasks.Task<object?> GetFromJsonAsync(this System.Net.Http.HttpClient client, string requestUri, System.Type type, System.Text.Json.JsonSerializerOptions? options = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
All of these methods are extensions on HttpClient.  Any reason they couldn't be extensions on the base HttpMessageInvoker instead?  Is there an advantage to only extending the derived type?

Reply:
No reason, it just never crossed my mind. Putting them on `HttpMessageInvoker` makes sense, assuming this gives us enough API surface to implement them.

Reply:
A lot of what's here is heavily influenced by the original set of APIs we had (based on newtonsoft) https://github.com/aspnet/AspNetWebStack/blob/master/src/System.Net.Http.Formatting/HttpClientExtensions.cs

We should question stuff (like the above feedback) - its likely that just wasn't considered before.